src/librustc_resolve/resolve_imports.rs

```diff
@@ -189,7 +189,18 @@ impl<'a> NameResolution<'a> {
                     return None;
                 }
 
-                return Indeterminate;
+                let target_module = match directive.target_module.get() {
+                    Some(target_module) => target_module,
+                    None => return Some(Indeterminate),
+                };
+                let name = match directive.subclass {
+                    SingleImport { source, target, .. } if source == target => target,
```

Comment:
I don't quite follow this -- why are you checking if `source == target`? Can you maybe give an example of some Rust code that would trigger this path?